pkg/hpa/service.go

```diff
@@ -765,3 +765,19 @@ func (c *Service) excludeExternalMetric(ctx context.Context, hpa *v2.HorizontalP
 
 	return newHPA
 }
+
+func (c *Service) checkHpaMetricStatus(ctx context.Context, currenthpa *v2.HorizontalPodAutoscaler, now time.Time) bool {
+	currenthpa = currenthpa.DeepCopy()
+	conditions := currenthpa.Status.Conditions
+
+	if conditions[1].Type == "ScalingActive" && conditions[1].Status == "True" {
+		//switch to Auto mode since metrics are back
+		return true
+	}
+
+	if conditions[1].Type == "ScalingActive" && conditions[1].Status == "False" && conditions[1].Reason == "FailedGetResourceMetric" {
```

Comment:
> FailedGetResourceMetric

What about other failures? e.g., the container resource metrics

Reply:
@sanposhiho should we only switch to emergency mode when main container resource metrics are missing? e.g. if istio-proxy metrics are unavailable for some reason but the main container metrics are still available, do we stay in auto mode since scaling is still active?

Reply:
I think we should fallback to emergency only if all metrics are dead..? Maybe it's too aggressive to do a fallback if some, but not all metrics are dead.

Reply:
e.g., Let's say a user deploys a new version to the app container which contains a bug unfortunately, and all main containers are crash due to the bug. In this case, switching to emerge doesn't help, rather just become unnecessary cost increase.

So, ideally we have to detect which issue is solvable with increasing the replicas, and which isn't.
I know it's super difficult, though. But, if two container's metrics are missing (a metric server is dead, the service is overwhelmed and whole dead, etc), we can say increasing the replica might help solve the issue, with more confidence.